btr-api/sample_data/generate_data.py

```diff
@@ -0,0 +1,228 @@
+            'startDate': date,
+        }
+        if interest['type'] in ['shareholding', 'votingRights']:
+            interest['share'] = { 'exact': fake.random_int(25, 100) }
+            interest['details'] = 'controlType.sharesOrVotes.registeredOwner'
+        else:
+            interest['details'] = 'controlType.other'
+
+        ooc_interests.append(interest)
+
+    return ooc_interests
+
+
+def _get_ooc_stmnts(entity_stmnt, person_stmnts):
+    """Return the ooc statements."""
+    ooc_stmnts: list[dict] = []
+    for person in person_stmnts:
+        ooc_stmnts.append({
+            'statementID': 'sample-ooc-' + fake.uuid4(),
+            'statementType': 'ownershipOrControlStatement',
+            'statementDate': entity_stmnt['statementDate'],
+            'isComponent': fake.random_element([True, False]),
+            'subject': {'describedByEntityStatement': entity_stmnt['statementID']},
+            'interestedParty': {'describedByPersonStatement': person['statementID']},
+            'publicationDetails': entity_stmnt['publicationDetails'],
+            'interests': _get_ooc_interests(entity_stmnt['statementDate'])
+        })
+
+    return ooc_stmnts
+
+
+def _get_entity_stmnt(statement_date: str):
+    """Return the entity statements."""
+    return {
+        'statementID': 'sample-entity-' + fake.uuid4(),
+        'statementType': fake.random_element(['business', 'branch']),
+        'statementDate': statement_date, 
+        'isComponent': fake.random_element([True, False]),
+        'entityType': 'legalEntity',
+        'name': fake.company(),
+        'identifiers': [{'id': f"TST-E{fake.uuid4()}", 'scheme': 'TST-E'}],
+        'publicationDetails': {
+            "publicationDate":  statement_date,
+            "bodsVersion": "1.0",
+            "publisher": {
+                "uri": "http://publisher.uri",
+                "name": "Publisher Name",
+                "source": {
+                "url": "http://source.url"
+                }
+            }
+        }
+    }
+
+
+def _get_person_stmnts(statement_date: str, locale: str, num: int = 1):
+    """Return the person statements."""
+    person_stmnts: list[dict] = []
+
+    def get_random_email():
+        """Return a randomly generated email."""
+        return (fake.name()).replace(' ', '_') + '@test.com'
+    
+    def get_country_from_locale():
+        """Return the country object based on the given locale."""
+        return pycountry.countries.get(alpha_2=locale.split('_')[1])
+
+    for _ in range(num):
+        name = {
+            'type': fake.random_element(['legal', 'knownAs', 'previous']),
+            'fullName': fake.name(),
+            'familyName': fake.last_name(),
+            'givenName': fake.first_name()
+        }
+
+        address_country = get_country_from_locale()
+
+        address = {
+            'street': fake.street_address(),
+            'city': fake.city(),
+            'country': {
```

Comment:
I have seen this after approved it :P

We are planning to have addresses flattened when storing them. So `country: {alpha_2, name}` will turn into country and `country` and `countryName`

e.g. 
```
{
  street: "fake street",
  city: "ontario",
  country: {
    alpha_2: "ca",
    name: "Canada"
  }
}
```
will become: 
```
{
  street: "fake street",
  city: "ontario",
  country: "ca",
  countryName: "Canada"
}
```

Also more info in this ticket:
https://github.com/bcgov/entity/issues/19919